### PR TITLE
Fixed the config factory naming glitch (issue 1720) 

### PR DESCRIPTION
**Pull request by niekraaijmakers**
Issue 1720

### CODE REVIEW

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestCookieHttpCacheConfigExtension.java [hunk 1]
```diff
@@ -43,10 +47,38 @@
...
+            description = "Defined key / values that will be allowed for this extension.")
+    public @interface Config {
+        @AttributeDefinition(
+                name = "Allowed keys",
+                description = "ValueMap keys that will used to generate a cache key."
+        )
+        String[] allowedKeys() default {};
+
+        @AttributeDefinition(
+                name = "AllowedValues",
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestCookieHttpCacheConfigExtension.java]:
```suggestion
                name = "Allowed values",
```

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestHeaderHttpCacheConfigExtension.java [hunk 2]
```diff
@@ -39,10 +44,38 @@
...
+            description = "Defined key / values that will be allowed for this extension.")
+    public @interface Config{
+        @AttributeDefinition(
+                name = "Allowed keys",
+                description = "ValueMap keys that will used to generate a cache key."
+        )
+        String[] allowedKeys() default {};
+
+        @AttributeDefinition(
+                name = "AllowedValues",
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestHeaderHttpCacheConfigExtension.java]:
```suggestion
                name = "Allowed values",
```

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestParameterHttpCacheConfigExtension.java [hunk 3]
```diff
@@ -40,9 +45,33 @@
...
+            description = "Defined key / values that will be allowed for this extension.")
+    public @interface Config{
+        @AttributeDefinition(
+                name = "Allowed keys",
+                description = "ValueMap keys that will used to generate a cache key."
+        )
+        String[] allowedKeys() default {};
+
+        @AttributeDefinition(
+                name = "AllowedValues",
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestParameterHttpCacheConfigExtension.java]:
```suggestion
                name = "Allowed values",
```

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/ValueMapValueHttpCacheConfigExtension.java [hunk 4]
```diff
@@ -42,9 +47,34 @@
...
+            description = "Defined key / values that will be allowed for this extension.")
+    public @interface Config{
+        @AttributeDefinition(
+                name = "Allowed keys",
+                description = "ValueMap keys that will used to generate a cache key."
+        )
+        String[] allowedKeys() default {};
+
+        @AttributeDefinition(
+                name = "AllowedValues",
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/ValueMapValueHttpCacheConfigExtension.java]:
```suggestion
                name = "Allowed values",
```

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestCookieHttpCacheConfigExtension.java [hunk 5]
```diff
@@ -43,10 +47,38 @@
...
-@Component(configurationPolicy = ConfigurationPolicy.REQUIRE, service = {HttpCacheConfigExtension.class, CacheKeyFactory.class})
-@Designate(ocd = KeyValueConfig.class, factory = true)
+@Component(
+        configurationPolicy = ConfigurationPolicy.REQUIRE,
+        service = {HttpCacheConfigExtension.class, CacheKeyFactory.class}
+)
+@Designate(ocd = RequestCookieHttpCacheConfigExtension.Config.class, factory = true)
 public class RequestCookieHttpCacheConfigExtension extends AbstractKeyValueExtension implements HttpCacheConfigExtension, CacheKeyFactory {
 
+    @ObjectClassDefinition(name = "ACS AEM Commons - HTTP Cache - RequestCookieHttpCacheConfigExtension - Key / Value extension",
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestCookieHttpCacheConfigExtension.java]:
```suggestion
    @ObjectClassDefinition(name = "ACS AEM Commons - HTTP Cache - RequestCookieHttpCacheConfigExtension",
```

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestHeaderHttpCacheConfigExtension.java [hunk 6]
```diff
@@ -39,10 +44,38 @@
...
-@Component(configurationPolicy = ConfigurationPolicy.REQUIRE, service = {HttpCacheConfigExtension.class, CacheKeyFactory.class})
-@Designate(ocd = KeyValueConfig.class, factory = true)
+@Component(
+        configurationPolicy = ConfigurationPolicy.REQUIRE,
+        service = {HttpCacheConfigExtension.class, CacheKeyFactory.class}
+)
+@Designate(ocd = RequestHeaderHttpCacheConfigExtension.Config.class, factory = true)
 public class RequestHeaderHttpCacheConfigExtension extends AbstractKeyValueExtension implements CacheKeyFactory, HttpCacheConfigExtension {
 
+    @ObjectClassDefinition(name = "ACS AEM Commons - HTTP Cache - RequestHeaderHttpCacheConfigExtension - Key / Value extension",
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestHeaderHttpCacheConfigExtension.java]:
```suggestion
    @ObjectClassDefinition(name = "ACS AEM Commons - HTTP Cache - RequestHeaderHttpCacheConfigExtension",
```

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestParameterHttpCacheConfigExtension.java [hunk 7]
```diff
@@ -40,9 +45,33 @@
  *
  */
 @Component(configurationPolicy = ConfigurationPolicy.REQUIRE, service = {HttpCacheConfigExtension.class, CacheKeyFactory.class})
-@Designate(ocd = KeyValueConfig.class, factory = true)
+@Designate(ocd = RequestParameterHttpCacheConfigExtension.Config.class, factory = true)
 public class RequestParameterHttpCacheConfigExtension extends AbstractKeyValueExtension implements CacheKeyFactory, HttpCacheConfigExtension {
 
+    @ObjectClassDefinition(name = "ACS AEM Commons - HTTP Cache - RequestParameterHttpCacheConfigExtension - Key / Value extension",
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/RequestParameterHttpCacheConfigExtension.java]:
```suggestion
    @ObjectClassDefinition(name = "ACS AEM Commons - HTTP Cache - RequestParameterHttpCacheConfigExtension",
```

File: bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/ValueMapValueHttpCacheConfigExtension.java [hunk 8]
```diff
@@ -42,9 +47,34 @@
  *
  */
 @Component(configurationPolicy = ConfigurationPolicy.REQUIRE, service = {HttpCacheConfigExtension.class, CacheKeyFactory.class})
-@Designate(ocd = KeyValueConfig.class, factory = true)
+@Designate(ocd = ValueMapValueHttpCacheConfigExtension.Config.class, factory = true)
 public class ValueMapValueHttpCacheConfigExtension extends AbstractKeyValueExtension implements CacheKeyFactory, HttpCacheConfigExtension {
 
+    @ObjectClassDefinition(name = "ACS AEM Commons - HTTP Cache - ValueMapValueHttpCacheConfigExtension - Key / Value extension",
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/httpcache/config/impl/ValueMapValueHttpCacheConfigExtension.java]:
```suggestion
    @ObjectClassDefinition(name = "ACS AEM Commons - HTTP Cache - ValueMapValueHttpCacheConfigExtension",
```

**Comment by niekraaijmakers** [comment]
All your changes are implemented david :)

**Comment by niekraaijmakers** [comment]
Hmm it broke due to some other unit test failing from other modules. can we try again?

**Comment by davidjgonzalez** [comment]
Closing as we reverted back to SCR annotations: https://github.com/Adobe-Consulting-Services/acs-aem-commons/issues/1741#issuecomment-461836942

**Pull request closed without merging**